File: code-snippets/node/refresh.js
const hubspot = require('@hubspot/api-client');
  
const hubspotClient = new hubspot.Client();

const GRANT_TYPES = {
  AUTHORIZATION_CODE: 'authorization_code',
  REFRESH_TOKEN: 'refresh_token',
};

const CLIENT_ID = 'your client id';
const CLIENT_SECRET = 'your client secret';

tokens = [{"accessToken":"accessToken1", "refreshToken":"refreshToken1"},
          {"accessToken":"accessToken2", "refreshToken":"refreshToken2"}];

const refreshToken = async (tokens) => {
    return Promise.all(tokens.map(async (token) => {
      const result = await hubspotClient.oauth.defaultApi.createToken(
        GRANT_TYPES.REFRESH_TOKEN,
        undefined,
        undefined,
        CLIENT_ID,
        CLIENT_SECRET,
        token.refreshToken,
    )
    return {
      ...result.body,
      updatedAt: Date.now()
    };
  })).then(response => console.log("API response on refresh:", response));
}

Promise.resolve(refreshToken(tokens));

File: python/code-snippets/refresh.py
from hubspot import HubSpot
client = HubSpot()

refresh_tokens_list = ['refresh_token1','refresh_token2', 'refresh_token3']

def refresh_tokens (refresh_tokens_list):
   new_tokens = []
   for refresh_token in refresh_tokens_list:
        new_token = client.auth.oauth.default_api.create_token(grant_type="refresh_token", redirect_uri="your redirect_uri", client_id = "your client id", client_secret = "your client secret", refresh_token = refresh_token)
        new_tokens.add(new_token)
   return new_tokens

new_tokens_list = refresh_tokens (refresh_tokens_list)
